src/storage/petastorm_storage_engine.py

```diff
@@ -0,0 +1,141 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+from src.spark.session import Session
+from src.catalog.models.df_metadata import DataFrameMetadata
+from petastorm.etl.dataset_metadata import materialize_dataset
+from src.storage.abstract_storage_engine import AbstractStorageEngine
+from src.utils.logging_manager import LoggingLevel
+from src.utils.logging_manager import LoggingManager
+from src.configuration.configuration_manager import ConfigurationManager
+
+from petastorm.unischema import dict_to_spark_row
+from petastorm.predicates import in_set, in_lambda
+from petastorm import make_reader
+from typing import Iterator, Dict
+
+
+class PetastormStorageEngine(AbstractStorageEngine):
+
+    def __init__(self):
+        """
+        Maintain a long live spark session and context.
+        """
+        self._spark = Session()
+        self.spark_session = self._spark.get_session()
+        self.spark_context = self._spark.get_context()
+
+    def _spark_url(self, table: DataFrameMetadata) -> str:
```

Comment:
The DataFrameMetadata object has the file uri. So we won't need this function.
table.file_url will do the job

Reply:
table.file_url is the path to the local file. This gets a spark url. In this case it appends "file:///tmp/eva_datasets" ahead.
spark or petastorm on longer allows local file path without prefix (throw tons of warning).